### DISCUSSION TITLE
Dictionary transforms with different readers for images and labels

### DISCUSSION THREAD

**Discussion by ikatechis** [Project-MONAI/MONAILabel#1226] [Q&A]
Hello everyone,

I would like to read 3d cbct scans with a custom dicom reader and annotations in npz format.
Is it possible to use two different readers (and two different LoadImage instances) to load the labels and images and then use dictionary transforms for the subsequent transforms? 
What would be the best way to do this?
Thank you!

**Comment by ikatechis** [comment]
Just realized I should have posted this in tutorials but I cannot find how to transfer it!

**Comment by tangy5** [comment]
Hi @ikatechis , thanks. Yes, the question is more like how to load NPZ format images with LoadImage transform. 
You can create a discussion in MONAI Core repo or Tutorials. We will also look into this, as part of MONAI team.

Thanks.

**Comment by diazandr3s** [comment]
> Hello everyone,
> 
> I would like to read 3d cbct scans with a custom dicom reader and annotations in npz format. Is it possible to use two different readers (and two different LoadImage instances) to load the labels and images and then use dictionary transforms for the subsequent transforms? What would be the best way to do this? Thank you!

Hi @ikatechis,

If you're interested in using MONAI Label for this task, I'd suggest converting the images and labels into the structure/format for MONAI Label to work. Here is the structure to follow: https://youtu.be/wtiEe_jiUzg?t=3299

Images and labels can be nii, nii.gz or NRRD.

Hope this helps,
